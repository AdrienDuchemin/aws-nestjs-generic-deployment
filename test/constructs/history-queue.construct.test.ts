import { expect as expectCDK, SynthUtils, countResources, haveResource } from '@aws-cdk/assert'
import { Stack } from '@aws-cdk/core'
import { HistoryQueue } from '../../src/constructs/history/history-queue.construct'

describe('History', () => {
  let stack: Stack

  beforeAll(()=> {
    stack = new Stack()
    new HistoryQueue(stack, 'HistoryQueue', { })
  })

  it('should create one Queue', () => {
    expectCDK(stack).to(countResources("AWS::SQS::Queue", 1))
    expectCDK(stack).to(haveResource("AWS::SQS::Queue",{
      FifoQueue: true,
    }))
  })

  it('should create one Lambda', () => {
    expectCDK(stack).to(countResources("AWS::Lambda::Function", 1))
    expectCDK(stack).to(haveResource("AWS::Lambda::Function",{
      Handler: 'index.handler'
    }))
  })

  it('should create one Table', () => {
    expectCDK(stack).to(countResources("AWS::DynamoDB::Table", 1))
    expectCDK(stack).to(haveResource("AWS::DynamoDB::Table",{
      KeySchema: [{ 
        AttributeName: "id",
        KeyType: "HASH",
      }],
    }))
  })

  it('should not have changed', () => {
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot()
  })
})