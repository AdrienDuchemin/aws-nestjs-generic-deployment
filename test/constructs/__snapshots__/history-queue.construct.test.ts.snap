// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History should not have changed 1`] = `
Object {
  "Parameters": Object {
    "AssetParameters6e299eb822cce449ea66aea74a07269cec00aa22054e7820f61d319cb0031e05ArtifactHash6241674D": Object {
      "Description": "Artifact hash for asset \\"6e299eb822cce449ea66aea74a07269cec00aa22054e7820f61d319cb0031e05\\"",
      "Type": "String",
    },
    "AssetParameters6e299eb822cce449ea66aea74a07269cec00aa22054e7820f61d319cb0031e05S3Bucket3765D8AA": Object {
      "Description": "S3 bucket for asset \\"6e299eb822cce449ea66aea74a07269cec00aa22054e7820f61d319cb0031e05\\"",
      "Type": "String",
    },
    "AssetParameters6e299eb822cce449ea66aea74a07269cec00aa22054e7820f61d319cb0031e05S3VersionKey59FE3835": Object {
      "Description": "S3 key for asset version \\"6e299eb822cce449ea66aea74a07269cec00aa22054e7820f61d319cb0031e05\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "HistoryQueueHistoryQueuefifo9642A4FD": Object {
      "Properties": Object {
        "FifoQueue": true,
        "QueueName": "HistoryQueueHistoryQueue.fifo",
      },
      "Type": "AWS::SQS::Queue",
    },
    "HistoryQueueHistoryQueuefifoHistoryQueueHistoryQueueHistoryLambdaCD0CD128": Object {
      "DependsOn": Array [
        "HistoryQueueHistoryQueuefifoHistoryQueueHistoryQueueHistoryLambdaServiceRoleDefaultPolicyC3E68B31",
        "HistoryQueueHistoryQueuefifoHistoryQueueHistoryQueueHistoryLambdaServiceRole8684B095",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters6e299eb822cce449ea66aea74a07269cec00aa22054e7820f61d319cb0031e05S3Bucket3765D8AA",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters6e299eb822cce449ea66aea74a07269cec00aa22054e7820f61d319cb0031e05S3VersionKey59FE3835",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters6e299eb822cce449ea66aea74a07269cec00aa22054e7820f61d319cb0031e05S3VersionKey59FE3835",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "TABLE_NAME": Object {
              "Ref": "HistoryQueueHistoryQueuefifoHistoryQueueHistoryQueueHistoryTableDEE80027",
            },
          },
        },
        "FunctionName": "HistoryQueueHistoryLambda",
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "HistoryQueueHistoryQueuefifoHistoryQueueHistoryQueueHistoryLambdaServiceRole8684B095",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Timeout": 60,
      },
      "Type": "AWS::Lambda::Function",
    },
    "HistoryQueueHistoryQueuefifoHistoryQueueHistoryQueueHistoryLambdaServiceRole8684B095": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "HistoryQueueHistoryQueuefifoHistoryQueueHistoryQueueHistoryLambdaServiceRoleDefaultPolicyC3E68B31": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "sqs:ReceiveMessage",
                "sqs:ChangeMessageVisibility",
                "sqs:GetQueueUrl",
                "sqs:DeleteMessage",
                "sqs:GetQueueAttributes",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "HistoryQueueHistoryQueuefifo9642A4FD",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": Array [
                "dynamodb:BatchGetItem",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
                "dynamodb:Query",
                "dynamodb:GetItem",
                "dynamodb:Scan",
                "dynamodb:BatchWriteItem",
                "dynamodb:PutItem",
                "dynamodb:UpdateItem",
                "dynamodb:DeleteItem",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "HistoryQueueHistoryQueuefifoHistoryQueueHistoryQueueHistoryTableDEE80027",
                    "Arn",
                  ],
                },
                Object {
                  "Ref": "AWS::NoValue",
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "HistoryQueueHistoryQueuefifoHistoryQueueHistoryQueueHistoryLambdaServiceRoleDefaultPolicyC3E68B31",
        "Roles": Array [
          Object {
            "Ref": "HistoryQueueHistoryQueuefifoHistoryQueueHistoryQueueHistoryLambdaServiceRole8684B095",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "HistoryQueueHistoryQueuefifoHistoryQueueHistoryQueueHistoryLambdaSqsEventSourceHistoryQueueHistoryQueuefifoCA8764EA": Object {
      "Properties": Object {
        "EventSourceArn": Object {
          "Fn::GetAtt": Array [
            "HistoryQueueHistoryQueuefifo9642A4FD",
            "Arn",
          ],
        },
        "FunctionName": Object {
          "Ref": "HistoryQueueHistoryQueuefifoHistoryQueueHistoryQueueHistoryLambdaCD0CD128",
        },
      },
      "Type": "AWS::Lambda::EventSourceMapping",
    },
    "HistoryQueueHistoryQueuefifoHistoryQueueHistoryQueueHistoryTableDEE80027": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "AttributeDefinitions": Array [
          Object {
            "AttributeName": "id",
            "AttributeType": "S",
          },
        ],
        "KeySchema": Array [
          Object {
            "AttributeName": "id",
            "KeyType": "HASH",
          },
        ],
        "ProvisionedThroughput": Object {
          "ReadCapacityUnits": 5,
          "WriteCapacityUnits": 5,
        },
        "TableName": "HistoryQueueHistoryTable",
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;
